lib/snapshot.js

```diff
@@ -379,31 +379,42 @@ function saveAsUnsupportedContent( filename, content_type ) {
 
 }
 
+/**
+ * Run a job from the queue if there is any, and the worker is not busy.
+ */
 function check_site_queue() {
-	if ( false === downloading ) {
-		if ( reload_config ) {
-			reload_config = false;
-			_blacklist.loadConfig();
-			logger.debug( process.pid + ': blacklist file reloaded' );
-		}
-		if ( site_queue.length > 0 ) {
-			downloading = true;
-			logger.debug( process.pid + ': site queue size = ' + site_queue.length );
-
-			var s_details = site_queue[0];
-			site_queue.shift();
-			logger.debug( process.pid + ': snapping: ' + s_details.url );
-			take_snapshot(
-				s_details.url,
-				s_details.file,
-				s_details.width,
-				s_details.height,
-				s_details.screenWidth,
-				s_details.screenHeight,
-				s_details.scaleFactor
-			);
-		}
+	if ( downloading || site_queue.length === 0 ) {
+		return;
 	}
+
+	if ( reload_config ) {
+		reload_config = false;
+		_blacklist.loadConfig();
+		logger.debug( process.pid + ': blacklist file reloaded' );
+	}
+
+	downloading = true;
+	logger.debug( process.pid + ': site queue size = ' + site_queue.length );
+
+	let s_details = site_queue.shift();
+
+	if ( s_details.action === 'analyze' ) {
+		logger.debug( process.pid + ': analyzing: ' + s_details.url );
+		_analyze_url.analyze( s_details.url, s_details.key );
+	} else {
+		logger.debug( process.pid + ': snapping: ' + s_details.url );
+		take_snapshot(
+			s_details.url,
+			s_details.file,
+			s_details.width,
+			s_details.height,
+			s_details.screenWidth,
+			s_details.screenHeight,
+			s_details.scaleFactor
+		);
+	}
+
+	downloading = false;
```

Comment:
I think it makes sense that this would be set to `false` here 👍  but why was this not done before? 

Actually with this added, It looks like you can clean up the three or so places in `take_snapshot()` where they manually reset `downloading` and `time_downloading`.
```
  time_downloading = 0;
  downloading = false;
```

Reply:
Valid point. There are a lot of things that can be refactored in the codebase of mShots. For now i made some changes to improve this and the next comment you wrote: 223814131ba981429617c6d06e1d9c78347cab05